sync/src/relayer/transactions_process.rs

```diff
@@ -0,0 +1,180 @@
+use crate::relayer::compact_block::{RelayTransaction, RelayTransactions};
+use crate::relayer::Relayer;
+use ckb_logger::debug_target;
+use ckb_network::{CKBProtocolContext, PeerIndex};
+use ckb_protocol::RelayTransactions as FbsRelayTransactions;
+use ckb_store::ChainStore;
+use failure::Error as FailureError;
+use fnv::FnvHashSet;
+use futures::{self, future::FutureResult, lazy};
+use std::convert::TryInto;
+use std::sync::Arc;
+use std::time::Duration;
+
+const DEFAULT_BAN_TIME: Duration = Duration::from_secs(3600 * 24 * 3);
+
+pub struct TransactionsProcess<'a, CS> {
+    message: &'a FbsRelayTransactions<'a>,
+    relayer: &'a Relayer<CS>,
+    nc: Arc<dyn CKBProtocolContext + Sync>,
+    peer: PeerIndex,
+}
+
+impl<'a, CS: ChainStore + Sync + 'static> TransactionsProcess<'a, CS> {
+    pub fn new(
+        message: &'a FbsRelayTransactions,
+        relayer: &'a Relayer<CS>,
+        nc: Arc<CKBProtocolContext + Sync>,
+        peer: PeerIndex,
+    ) -> Self {
+        TransactionsProcess {
+            message,
+            relayer,
+            nc,
+            peer,
+        }
+    }
+
+    pub fn execute(self) -> Result<(), FailureError> {
+        let relay_txs: RelayTransactions = (*self.message).try_into()?;
+
+        let txs: Vec<RelayTransaction> = {
+            let tx_filter = self.relayer.shared().tx_filter();
+
+            relay_txs
+                .transactions
+                .into_iter()
+                .filter(|relay_tx| {
+                    let tx_hash = relay_tx.transaction.hash();
+                    if tx_filter.contains(tx_hash) {
+                        debug_target!(
+                            crate::LOG_TARGET_RELAY,
```

Comment:
Ditto

Reply:
Updated in the last commit, please review again, thanks.